container/shim/src/index.js

```diff
@@ -268,7 +288,21 @@ app.post('/data/:cid', function (req, res) {
 
 addRegisterCheckRoute(app)
 
-export default app
+function proxyRequestHeaders (reqHeaders) {
+  const headers = { 'User-Agent': NODE_UA }
+
+  for (const key of PROXY_REQUEST_HEADERS) {
+    if (key in reqHeaders) {
+      if (key === 'x-if-none-match') {
+        headers['if-none-match'] = reqHeaders[key]
```

Comment:
Would there be value in writing this generic for all `x-*` headers?

Reply:
Eh I thought about it, but then we'd proxy headers that shouldn't be proxied, like these two https://github.com/filecoin-saturn/L1-node/blob/c74ebd4359d065b08d047c2dd3df7e4b0a57f386/container/nginx/confs/proxy.tlsconf#L84-L85

This is a hacky workaround since nginx swallows the `if-none-match` header, if we need to forward additional headers in this manner then we can come up with something generic.